pyboy/core/cartridge/mbc1.py

```diff
@@ -42,6 +28,28 @@ def setitem(self, address, value):
                                "banks are not initialized. Initializing %d RAM banks as "
                                "precaution" % (value, address, EXTERNAL_RAM_TABLE[0x02]))
                 self.init_rambanks(EXTERNAL_RAM_TABLE[0x02])
-            self.rambanks[self.rambank_selected][address-0xA000] = value
+
+            if self.rambank_enabled:
+                self.rambank_selected = self.bank2 if self.memorymodel else 0
+                self.rambanks[self.rambank_selected % len(self.rambanks)][address-0xA000] = value
         else:
             raise logger.error("Invalid writing address: %s" % hex(address))
+
+    def getitem(self, address):
+        if 0x0000 <= address < 0x4000:
+            self.rombank_selected = (self.bank2 << 5) if self.memorymodel else 0
```

Comment:
I like being explicit about boolean conversions. It tells more about your intentions. What value do you assume `self.memorymodel` is? Should it be `self.memorymodel != 0`, `self.memorymodel > 0` or always `self.memorymodel == 1`?